core/client/src/main/java/alluxio/client/block/LocalBlockInStream.java

```diff
@@ -158,6 +158,11 @@ protected void incrementBytesReadMetric(int bytes) {
     Metrics.BYTES_READ_LOCAL.inc(bytes);
   }
 
+  @Override
+  boolean isLocal() {
+    return true;
```

Comment:
Using `instanceof` should work too, but I think it's usually better to avoid reflection where possible. This way it's explicit that block in streams should report whether they are local, so that if a new type of local stream is added, or a mock stream is used for testing, it will be easy to see that it needs to report whether it's local.

We want to be able to ask any `BlockInStream` whether it is local or not, but we can't use `Locatable` because we don't know the address of an `UnderStoreBlockInStream`